### ISSUE TITLE
hoogle.haskell.org returns 502: Bad Gateway

### ISSUE THREAD

**Issue by ulysses4ever** [ndmitchell/hoogle#291]

**Comment by ulysses4ever** [comment]
Oh, this appears to be some local network quirk. Sorry for the noise.

**Comment by ndmitchell** [comment]
It does this for a few minutes every day while it updates itself. I should probably move this to the dead time zone of the world (after US goes to sleep), and reduce the gap.

**Comment by ulysses4ever** [comment]
Good to know, thanks!

**Comment by ndmitchell** [comment]
I optimised the log parsing, so the gap itself should be 5x less. I also moved the restart to 3am UK time. The original motivation was that if it upgraded and keeled over, I was more likely to be around about 8pm. That has never happened, and Hoogle logs show 3am GMT as the quietest period, so scheduling for then.

**Comment by joprice** [comment]
I just hit this. I don't know anything about the hoogle backend, but could this be worked around by keeping the old version up until the new one is ready? I'd be interested in helping if there's a way to work around it.

**Comment by ulysses4ever** [comment]
Yes, living on the East Coast and doing some late night codings occasionally, I regularly ran into this too, unfortunately. Keeping the old version until the actual switch is possible momentarily would be great.

**Comment by ndmitchell** [comment]
Current the process is:

* Compile and test a new release.
* Shut down the old one.
* Start up the new one.
* New one reads the log file to produce summary data.
* New one starts serving and writing to the logs.

The two options are:

* Don't shut the old one until the new one responds with a heartbeat. Probably requires sysadmin stuff to switch the ports over?
* Make the log parsing lazy, so Hoogle can respond before the logs are parsed.

I'd accept patches for each of them - I certainly agree that having a hiccup every day isn't good.

**Comment by knitHacker** [comment]
This is happening to me right now and it is not 3am in the UK.
